R/sircovid.R

```diff
@@ -1,7 +1,127 @@
+##' 
+##' @param obs_params List of parameters used for comparing
+##'   model to data in the particle filter
+##'   
+##' @param n_particles Number of particles  
+##' 
+##' @returns Results from particle filter
+##' 
+##' @export
+run_particle_filter <- function(case_data_file,
+                                admissions_data_file,
+                                itu_colname = "number_of_confirmed_covid_19_patients_in_hdu_itu_at_0800_total",
+                                itu_ignore_dates = c("2020-03-15"),
+                                first_data_date = "2020-03-01",
+                                model_start_date = "2020-02-02",
+                                country_name = "United_Kingdom",
+                                case_file_day_shift = 2, #####SHIFT BY 1 day or 2?
+                                model_params,
+                                obs_params = list(phi_ICU = 0.95,
+                                                  k_ICU = 2,
+                                                  phi_death = 1789/1651,
+                                                  k_death = 2,
+                                                  exp_noise = 1e6),
+                                n_particles = 1000) {
+  
+  # parameter checks
+  if (as.Date(first_data_date, "%Y-%m-%d") < as.Date(model_start_date, "%Y-%m-%d")) {
+    stop("Model start date is later than data start date")
+  }
+  
+  # Read death data file, and manipulate. Select country only, 
+  # and ensure dates match up
+  case_data <- read.csv(case_data_file, sep = ",")
+  case_data$dateRep <- as.Date(case_data$dateRep, format="%d/%m/%Y") - case_file_day_shift
+  case_data <- case_data[case_data$dateRep >= as.Date(first_data_date),]
+  case_data <- case_data[case_data$countriesAndTerritories == country_name,]
+  if (nrow(case_data) < 1) {
+    stop("No case data matching country and date filtering criteria")
+  }
+  
+  # Open a time series and format
+  # these are by task XX on ncov-outputs
+  admissions_data <- read.csv(admissions_data_file)
+  if (!(itu_colname %in% names(admissions_data))) {
+    stop("itu_colname not found in time_series_file")
+  } else {
+    names(admissions_data)[names(admissions_data) == itu_colname] <- "itu"
+  }
+  admissions_data$date <- as.Date(admissions_data$date, format="%Y-%m-%d")
+  admissions_data <- admissions_data[admissions_data$date >= as.Date(first_data_date),]
+  if (nrow(admissions_data) < 1) {
+    stop("No admissions data matching date filtering criteria")
+  }
+  
+  # Merge mortality and admissions data, allowing NAs where there
+  # is no overlap
+  data <- merge(admissions_data, case_data, 
+                by.x="date", by.y="dateRep",
+                all.x=TRUE, all.y=TRUE)[,c("date", "deaths", "itu")]
+  if (nrow(data) < 1 || 
+      nrow(data) == nrow(admissions_data) + nrow(case_data) || 
+      ncol(data) != 3) {
+    stop("Merge of case and admissions data failed - do dates match?")
+  }
```

Comment:
@OJWatson is reconstructing the times series in his function for sampling the grid, we might want to integrate the two approaches later.

Reply:
https://github.com/mrc-ide/sircovid/blob/master/R/grid_search.R#L88-L98 could eventually be replaced by this function then?